Valence new

EvienCheng opened the pull request:
## Description
Provide a brief description of the PR's purpose here.

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [ ] TODO 1

## Questions
- [ ] Question1

## Status
- [ ] Ready to go

dependabot[bot] committed:
Bump mamba-org/setup-micromamba from 1 to 2 (#73)

Bumps [mamba-org/setup-micromamba](https://github.com/mamba-org/setup-micromamba) from 1 to 2.
- [Release notes](https://github.com/mamba-org/setup-micromamba/releases)
- [Commits](https://github.com/mamba-org/setup-micromamba/compare/v1...v2)

---
updated-dependencies:
- dependency-name: mamba-org/setup-micromamba
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mattwthompson committed:
Use Interchange 0.4 (#76)

* Update test environment

* Update to use new annotated types

* Fix type-checking

* Slightly simplify diff

* Use `macos-latest`

* Bump codecov/codecov-action from 4 to 5

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 4 to 5.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v4...v5)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update .github/workflows/ci.yaml

* Apply suggestions from code review

Co-authored-by: Josh A. Mitchell <yoshanuikabundi@gmail.com>

* Cleanup

* Relax RDKit pin, trust toolkit metadata?

* Add Python 3.12

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Josh A. Mitchell <yoshanuikabundi@gmail.com>

mattwthompson committed:
Drop testing on Python 3.10 (#81)

fjclark committed:
Add Urey-Bradley plug-in (#82)

* Add Urey-Bradley plug-in

* Specify atom positions in Urey-Bradley test

Removes dependence on conformer generation. Also,
fix a typo (UreyBradley should have been UreyBradleys).

* Return only relevant atoms from valence_terms in Urey-Bradley collection

e.g. return atoms i and k rather than all atoms in the angle (i, j, and
k).

* Explicitly name the Urey-Bradley OpenMM force

This simplifies the tests by allowing the UB force to be easily
identified.

* Check we get expected number of OMM indices when creating UB force

fjclark committed:
Remove spaces from the Urey-Bradley energy expression (#83)

This makes the expression identical to the harmonic bond forces and
streamlines working with SMEE.

EvienCheng committed:
changes to vsites.py

EvienCheng committed:
Merge remote-tracking branch 'upstream/main'

EvienCheng committed:
Added functional forms to new valence file

pavankum commented on smirnoff_plugins/collections/valence.py:
```diff
@@ -114,3 +114,213 @@ def modify_openmm_forces(
+class SMIRNOFFLeeKrimmCollection(SMIRNOFFCollection):
+    type: ClassVar[Literal["LeeKrimm"]] = "LeeKrimm"
+    is_plugin: ClassVar[bool] = True
+    acts_as: str = "ImproperTorsions"
```
Just a style comment, make this consistent with the handler name.

pavankum commented on smirnoff_plugins/collections/valence.py:
```diff
@@ -114,3 +114,213 @@ def modify_openmm_forces(
+class SMIRNOFFTwoMinimaCollection(SMIRNOFFCollection):
+    type: ClassVar[Literal["TwoMinima"]] = "TwoMinima"
+    is_plugin: ClassVar[bool] = True
+    acts_as: str = "ImproperTorsions"
```
Same thing here.

pavankum commented on smirnoff_plugins/collections/valence.py:
```diff
@@ -114,3 +114,213 @@ def modify_openmm_forces(
+from smirnoff_plugins.handlers.valence import LeeKrimmHandler
+
+class SMIRNOFFLeeKrimmCollection(SMIRNOFFCollection):
+    type: ClassVar[Literal["LeeKrimm"]] = "LeeKrimm"
```
What does ClassVar do here, I might be missing something, what's its purpose?

EvienCheng committed:
added harmonic angle

EvienCheng committed:
update unique parameters

EvienCheng committed:
removed extra files

pavankum committed:
cleaned up handlers, removed two minima form

pavankum committed:
remove ClassVar, reformat with black, isort, flake8

pavankum committed:
add back find_matches(unique)

pavankum committed:
revised new fn form implementation;added tests;made names singular, UreyBradley also got affected

EvienCheng committed:
added find_matches back in

pavankum committed:
Revert findmatches, and claude review of code

pavankum committed:
conflicts resolved

pavankum committed:
resolving conflicts

pavankum committed:
Remove leftover conflict markers and dead bonded.py entry points

A prior conflict resolution against an older upstream Urey-Bradley
commit was committed without actually resolving it, leaving literal
<<<<<<</=======/>>>>>>> markers in valence.py (collections/handlers),
its tests, and setup.py. Resolved in favor of the singular UreyBradley
naming and the HarmonicHeight/LeeKrimm/HarmonicAngle collections
already present on this branch.

Also drops the duplicate HarmonicHeightHandler/Collection entry points
pointing at smirnoff_plugins.handlers.bonded, which doesn't exist
(collections/bonded.py's import of it was already broken and unused).
